include/crocoddyl/core/activations/quadratic-flat.hpp

```diff
@@ -0,0 +1,75 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2019, LAAS-CNRS
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_CORE_ACTIVATIONS_QUADRATIC_FLAT_HPP_
+#define CROCODDYL_CORE_ACTIVATIONS_QUADRATIC_FLAT_HPP_
+
+#include <stdexcept>
+#include "crocoddyl/core/fwd.hpp"
+#include "crocoddyl/core/activation-base.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+#include <iostream>
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class ActivationModelQuadFlatTpl : public ActivationModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef ActivationModelAbstractTpl<Scalar> Base;
+  typedef ActivationDataAbstractTpl<Scalar> ActivationDataAbstract;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+
+  explicit ActivationModelQuadFlatTpl(const std::size_t& nr,const Scalar& sigma2)
+     : Base(nr), sigma2_(sigma2){};
+  virtual ~ActivationModelQuadFlatTpl(){};
+
+  virtual void calc(const boost::shared_ptr<ActivationDataAbstract>& data, const Eigen::Ref<const VectorXs>& r) {
+    if (static_cast<std::size_t>(r.size()) != nr_) {
+      throw_pretty("Invalid argument: "
+                   << "r has wrong dimension (it should be " + std::to_string(nr_) + ")");
+    }
+    data->a_value = (Scalar(1.0) - exp((-r.transpose() * r)[0] / sigma2_));
+  };
+
+  virtual void calcDiff(const boost::shared_ptr<ActivationDataAbstract>& data, const Eigen::Ref<const VectorXs>& r) {
+    if (static_cast<std::size_t>(r.size()) != nr_) {
+      throw_pretty("Invalid argument: "
+                   << "r has wrong dimension (it should be " + std::to_string(nr_) + ")");
+    }
+    a0_ = Scalar(2.0) / sigma2_ * exp((-r.transpose() * r)[0] / sigma2_);
```

Comment:
same here

Reply:
To avoid duplication of computations, I suggest to create a data for ```exp((-r.transpose() * r)[0] / sigma2_``` and then reuse the value computed in `calc` here.